desktop-ui/emulator/nintendo-64.cpp

```diff
@@ -209,107 +212,167 @@ auto Nintendo64::load(Menu menu) -> void {
 
 auto Nintendo64::portMenu(Menu& portMenu, ares::Node::Port port) -> void {
   if(port->type() != "Controller") return;
+  auto peripheral = port->connected();
+  // Only show Pak menu if a Gamepad is connected
+  if(!peripheral || peripheral->name() != "Gamepad") return;
+  
+  // Check what pak is currently connected
+  ares::Node::Peripheral pak = nullptr;
+  if(auto pakPort = peripheral->find<ares::Node::Port>("Pak")) {
+    pak = pakPort->connected();
+  }
 
   const string portNum = port->name()[port->name().length() - 1];
 
-  // remove this check to enable pak menu option for all 4 controllers
   if(portMenu.actionCount() > 0) portMenu.append(MenuSeparator());
   Menu pakMenu{&portMenu};
   pakMenu.setText("Pak");
   Group pakGroup;
+
+  // Show Nothing option
   MenuRadioItem nothing{&pakMenu};;
   nothing.setText("Nothing");
   nothing.setAttribute<ares::Node::Port>("port", port);
+  if(!pak) nothing.setChecked();
   nothing.onActivate([=] {
     auto port = nothing.attribute<ares::Node::Port>("port");
     const string portName = port->name();
     if(auto port = emulator->root->find<ares::Node::Port>(portName)) {
-      port->disconnect();
```

Comment:
This pattern of re-creating the Gamepad was preventing the ability to report the "Accessory Changed" status. This menu is only shown when a Gamepad is selected, so re-creating it is not strictly necessary.

LibDragon's Joypad subsystem cannot detect a controller disconnect if another controller is connected in the same frame.

Similarly, Joypad subsystem accessory re-detection relies on either:
1. Controller reconnect is reported, which requires at least one frame between unplugging the current controller and plugging in the new one.
2. `0x02` (Pak Absent) status is reported, which requires at least one frame between removing the current accessory and connecting the new one.
3. `0x03` (Pak Changed) status is reported, which is required if swapping the accessory instantly.